app/src/main/java/com/csc/lesson3/ResultActivity.java

```diff
@@ -0,0 +1,100 @@
+package com.csc.lesson3;
+
+import android.content.res.Configuration;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.TextView;
+import com.csc.lesson3.services.BingImages;
+import com.csc.lesson3.services.YandexTranslator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ResultActivity extends MainActivity {
+
+    public static final String INPUT = "INPUT";
+    private static final String STATE_URLS = "STATE_URLS";
+    private static final String STATE_TRANSLATION = "STATE_TRANSLATION";
+    private String input_cache;
+
+    TextView translation;
+    RecyclerView recyclerView;
+    RVAdapter adapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        recyclerView = (RecyclerView)findViewById(R.id.rv);
+        GridLayoutManager gridLayoutManager = new GridLayoutManager(this, this.getResources().getConfiguration().orientation== Configuration.ORIENTATION_PORTRAIT ? 3 : 5);
+        recyclerView.setLayoutManager(gridLayoutManager);
+        adapter = new RVAdapter(this);
+        recyclerView.setAdapter(adapter);
+
+        translation = ((TextView) findViewById(R.id.translation));
+
+        if (savedInstanceState != null) {
+            adapter.updateUrls(savedInstanceState.getStringArrayList(STATE_URLS));
+        }
+        else {
+            String input = getIntent().getStringExtra(INPUT);
+            editText.setText(input);
+            getTranslation(input);
+            getImages(input);
+        }
+        input_cache = editText.getText().toString();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        editText.setText(input_cache);
```

Comment:
EditText have id:
android:id="@+id/input"

Also I check rotation rigth now, it's works.